src/posix/platform/hdlc_interface.cpp

```diff
@@ -330,6 +330,8 @@ void HdlcInterface::UpdateFdSet(fd_set &aReadFdSet, fd_set &aWriteFdSet, int &aM
     OT_UNUSED_VARIABLE(aWriteFdSet);
     OT_UNUSED_VARIABLE(aTimeout);
 
+    assert(mSockFd != -1);
```

Comment:
Why we need to add `assert()` here? If the `mSockF` equals to `-1` the program will die when calling the `Init()`.